maven-core/src/main/java/org/apache/maven/project/MavenProject.java

```diff
@@ -182,6 +183,18 @@ public class MavenProject implements Cloneable {
 
     private final Set<String> lifecyclePhases = Collections.synchronizedSet(new LinkedHashSet<>());
 
+    private final AtomicBoolean projectNeedsBuild = new AtomicBoolean(true);
```

Comment:
this should only be set at the beginning, why using an `AtomicBoolean` ?

Reply:
I wanted just to ensure that "transition" from true->false can happen only once, and there is no way to "reset" it

Reply:
Again, this is just a hack to prove my theory, this flag is omitted from clone etc...